taca/server_status/cronjobs.py

```diff
@@ -9,35 +9,42 @@
 
 from taca.utils.config import CONFIG
 
-def update_cronjob_db():
-    logging.info('Getting list of cronjobs')
-    crontab = CronTab()
-    # CronTab() is supposed to return the list of cronjobs
-    # it works on milou, but doesn't work on preproc (returns an empty list)
-    # when it doesn't work, create CronTab object from the output of command 'crontab -l'
-    if crontab.crons == []:
+def _parse_crontab():
+    result = {}
+    user = getpass.getuser()
+    logging.info('Getting crontab for user {}'.format(user))
+    crontab = None
+    try:
+        crontab = CronTab(user=user)
+    except Exception, e:
+        logging.warning('Cannot get a crontab for user: {}'.format(user))
+        logging.warning(e.message)
         try:
-            p = subprocess.Popen('crontab -l', shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            logging.info('Getting a crontab for default user')
+            crontab = CronTab()
```

Comment:
ah, that is strange. Cause when I tried it in `NAS-10` did not work for me

```
>>> from crontab import CronTab
>>> c = CronTab()
>>> c.crons
[]
```

However, `CronTab(user=True)` worked. Maybe difference in `crontab` package version ?

```
>>> c = CronTab(user=True)
>>> c.crons
[<CronJob '@reboot /usr/bin/supervisord -c /home/hiseq.bioinfo/config/supervisord.conf # Ansible: Start supervisord on reboot'>, <CronJob '@hourly /usr/sbin/logrotate /home/hiseq.bioinfo/config/logrotate.conf -s /home/hiseq.bioinfo/config/logrotate.status # Ansible: Run logrotate hourly'>, <CronJob '@daily source ~/.bashrc && taca storage cleanup --site nas --hours 18 # Ansible: Trigger automatic cleanup'>, <CronJob '0 1 * * * source ~/.bashrc && taca backup encrypt # Ansible: Encrypt HiSeq finished runs in nosync'>, <CronJob '*/15 * * * * source /srv/mfs/configs/scripts/check_lsync.sh # Check if the lsync process is running, if not send a mail'>]
```

Anyway, isn't this same as the one in first `try` (`crontab = CronTab(user=user)`) ? As the `user` is got from `getpass.getuser()` (which will also return the current user)